### DISCUSSION TITLE
Family Calendar Card

### DISCUSSION THREAD

**Discussion by Daniel100111** [piitaya/lovelace-mushroom#710] [Ideas]
May I please ask if it will be possibl to create mushroom Calendar card. Want to list the next two days events. (All day, and timed events). Something for the family to use.

**Comment by GizmoGoody** [comment]
I've done this using the Outlook 365 HACS integration (you would use whatever integration gets your calendar into HA...e.g. Google Calendar, etc.) and a markdown card.

```
            - type: markdown
              content: >-
                {% set cal = state_attr('calendar.calendar_family','data') %}
                {% for event in cal if as_timestamp(event.start) < as_timestamp(now())+states('input_select.timeframe_next_calendar_events')|int*60*60 -%}
                {% set start_time = event.start -%}
                {% set end_time = event.end -%}
                {% set formatted_start_time = (as_timestamp(start_time))|timestamp_custom ('%a %-I:%M %p') -%}
                {% set formatted_end_time = (as_timestamp(end_time))|timestamp_custom ('%-I:%M %p') -%}
                {% set formatted_day = (as_timestamp(start_time)+18000)|timestamp_custom ('%a') -%}
                  {% if event.all_day == True -%}
                  **{{ formatted_day }}**{{ " | " + event.summary -}}
                  {% else -%}
                  **{{ formatted_start_time }} - {{ formatted_end_time }}**{{ " | " + event.summary -}}
                  {% endif %}
                {% endfor %}
```

If you wanted the range to be static, you could just replace the `states('input_select.timeframe_next_calendar_events')|int*60*60` with 172800 (48 hours in seconds).

**Reply by Daniel100111** [reply]
Thank you so much for your help.

I've creared the card but getting it right. 

![image](https://user-images.githubusercontent.com/31239012/197569227-99b96342-1b5a-4510-ad2e-c4eb065b3f5e.png)

Any help will be much appriciated.

**Reply by GizmoGoody** [reply]
What's your question/issue exactly?

**Comment by Daniel100111** [comment]
I do not get the card to work. Getting a blank card as per picture. Really appriciate your help with this.

**Reply by GizmoGoody** [reply]
Copy/paste what I put into Developer Tools > Template, then change the Calendar entity to your calendar. Did it work? If not, what error did you get?

**Comment by Daniel100111** [comment]
TemplateSyntaxError: Unexpected end of template. Jinja was looking for the following tags: 'endfor' or 'else'. The innermost block that needs to be closed is 'for'.

**Comment by Daniel100111** [comment]
ValueError: Template error: int got invalid input 'unknown' when rendering template 'type: markdown
content: >-
 {% set cal = state_attr('calendar.calendar_daniel','data') %}
    {% for event in cal if as_timestamp(event.start) < as_timestamp(now())+states('input_select.timeframe_next_calendar_events')|int*60*60 -%}
    {% set start_time = event.start -%}
    {% set end_time = event.end -%}
    {% set formatted_start_time = (as_timestamp(start_time))|timestamp_custom ('%a %-I:%M %p') -%}
    {% set formatted_end_time = (as_timestamp(end_time))|timestamp_custom ('%-I:%M %p') -%}
    {% set formatted_day = (as_timestamp(start_time)+18000)|timestamp_custom ('%a') -%}
    {% if event.all_day == True -%}
      **{{ formatted_day }}**{{ " | " + event.summary -}}
      {% else -%}
      **{{ formatted_start_time }} - {{ formatted_end_time }}**{{ " | " + event.summary -}}
      {% endif %}
    {% endfor %}' but no default was specified

**Reply by GizmoGoody** [reply]
Make sure there is a * before each 60

if that still results in an error, try this line to make sure you’re getting all of the same attributes that I’m using:

{{state_attr('calendar.calendar_daniel','data') }}

You can also enter…

calendar.calendar_daniel

…under Developer Tools>States to see all of the available attributes, then modify as necessary
